bugsnag-plugin-android-ndk/build.gradle

```diff
@@ -10,6 +10,9 @@ apply plugin: "com.android.library"
 
 dependencies {
     api(project(":bugsnag-android-core"))
+    // newest version that didn't require fiddling with sourceCompatibility -
+    // didn't use kotlin module to avoid runtime version conflicts
+    androidTestImplementation("com.fasterxml.jackson.core:jackson-databind:2.12.5")
```

Comment:
It feels preferable to use dsl-json or GSON rather than bringing another JSON serializer into the project, even if it's just for test code

Reply:
e.g. 

```
byte[] input;
DslJson<Object> dslJson = new DslJson<Object>();
Map<String, Object> map = dslJson.deserialize(Map.class, input, input.length);
```

Reply:
I attempted to pull in GSON (since we only have the streaming bits vendored in), but its giving me some JSON validity warnings for reasons that don't seem worth chasing for tests. dsl-json does not (yet?) exist in the repo this juncture, but I can fix it after the next merge or somthing to use `JsonHelper`.

Reply:
Yeah - it'd be worth merging in `next` if the integration branch is behind, as that'd allow this to use `JsonHelper`.

Reply:
Its a non-trivial merge (with a few conflicts in `Event` classes), so it might be worth holding until I get the tests in. I've got tests for all of the subsequent migrations and JSON serialization ready to go.